src/domain/opendatafabric/src/identity/dataset_identity.rs

```diff
@@ -358,6 +376,27 @@ impl DatasetAlias {
     }
 }
 
+impl PartialEq<Self> for DatasetAlias {
+    fn eq(&self, other: &Self) -> bool {
+        ((self.account_name.is_none() && other.account_name.is_none())
+            || (self.account_name.is_some()
+                && other.account_name.is_some()
+                && self
+                    .account_name
+                    .as_ref()
+                    .unwrap()
+                    .lowercase_eq(other.account_name.as_ref().unwrap())))
+            && self.dataset_name.lowercase_eq(&other.dataset_name)
```

Comment:
could you please use early returns to split conditions?  it'll make it easier to read

Reply:
Due to lastest changes this block was removed